File: services/server/src/main/java/com/yaytsa/server/infrastructure/persistence/entity/ApiTokenEntity.java
package com.yaytsa.server.infrastructure.persistence.entity;

import jakarta.persistence.*;
import java.time.OffsetDateTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(
    name = "api_tokens",
    uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "device_id"}))
public class ApiTokenEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id", nullable = false)
  private UserEntity user;

  @Column(unique = true, nullable = false, length = 64)
  private String token;

  @Column(name = "device_id", nullable = false, length = 255)
  private String deviceId;

  @Column(name = "device_name", length = 255)
  private String deviceName;

  @CreationTimestamp
  @Column(name = "created_at", nullable = false, updatable = false)
  private OffsetDateTime createdAt;

  @Column(name = "last_used_at")
  private OffsetDateTime lastUsedAt;

  @Column(name = "expires_at")
  private OffsetDateTime expiresAt;

  @Column(nullable = false)
  private boolean revoked;

  public boolean isValid() {
    if (revoked) {
      return false;
    }
    if (expiresAt != null && OffsetDateTime.now().isAfter(expiresAt)) {
      return false;
    }
    return user != null && user.isActive();
  }
}

File: yay-tsa-v2/infra-persistence/auth/src/main/kotlin/dev/yaytsa/persistence/auth/entity/ApiTokenEntity.kt
package dev.yaytsa.persistence.auth.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "api_tokens", schema = "core_v2_auth")
class ApiTokenEntity(
    @Id
    @Column(name = "id", nullable = false)
    var id: UUID = UUID.randomUUID(),
    @Column(name = "user_id", nullable = false)
    var userId: UUID = UUID.randomUUID(),
    @Column(name = "token", nullable = false)
    var token: String = "",
    @Column(name = "device_id", nullable = false)
    var deviceId: String = "",
    @Column(name = "device_name")
    var deviceName: String? = null,
    @Column(name = "created_at", nullable = false)
    var createdAt: Instant = Instant.now(),
    @Column(name = "last_used_at")
    var lastUsedAt: Instant? = null,
    @Column(name = "expires_at")
    var expiresAt: Instant? = null,
    @Column(name = "revoked", nullable = false)
    var revoked: Boolean = false,
)
